grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java

```diff
@@ -71,17 +73,28 @@ public void setupSession(){
     }
 
     @Test
-    public void whenSingleQueryLoadedAndTaskCompletionFunctionThrowsError_ErrorIsLogged(){
+    public void whenSingleQueryLoadedAndTaskCompletionFunctionThrowsError_ErrorIsLogged() throws InterruptedException {
+        CountDownLatch callbackCompleted = new CountDownLatch(1);
+
         // Create a BatchMutatorClient with a callback that will fail
         BatchMutatorClient loader = loader();
-        loader.setTaskCompletionConsumer((json) -> assertTrue("Testing Log failure",false));
+        loader.setTaskCompletionConsumer((json) -> {
+            try {
+                fail("Testing failures appear in logs");
+            } finally {
+                callbackCompleted.countDown();
+            }
+        });
 
         // Load some queries
         generate(this::query).limit(1).forEach(loader::add);
 
         // Wait for queries to finish
         loader.waitToFinish();
 
+        // Wait for callback function to execute - Callbacks are run asynchronously
+        callbackCompleted.await(10, TimeUnit.SECONDS);
+
```

Comment:
I think you should check if callbackCompleted.await returns 'false' and fail the test if hat happens. You are expecting it to complete in 10s, but if it doesn't that would reveal a bug somewhere.